core/src/main/java/org/kohsuke/stapler/export/Property.java

```diff
@@ -199,118 +197,122 @@ private void writeValue(Type expected, Object value, TreePruner pruner, DataWrit
 
         Class c = value.getClass();
 
-        Model model;
-        try {
-            model = owner.get(c, parent.type, name);
-        } catch (NotExportableException ex) {
-            if(STRING_TYPES.contains(c)) {
-                writer.value(value.toString());
-                return;
+        if (c.getAnnotation(ExportedBean.class) == null) {
```

Comment:
Am I missing something or would this not be caught by the null return value from `getOrNull` a few lines later anyway?

Reply:
You are absolutely right. Fixed.